src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -262,7 +266,44 @@ private void initDragAndDrop() {
                                 if (libraryTab.getId().equals(tab.getId()) &&
                                         !tabbedPane.getSelectionModel().getSelectedItem().equals(libraryTab)) {
-                                    ((LibraryTab) libraryTab).dropEntry(stateManager.getLocalDragboard().getBibEntries());
+                                    LibraryTab libraryTabCasted = (LibraryTab) libraryTab;
+                                    if (dragboard.hasContent(DragAndDropDataFormats.GROUP)) {
+                                        List<String> groupPathToSources = (List<String>) dragboard.getContent(DragAndDropDataFormats.GROUP);
+                                        if (libraryTabCasted.getBibDatabaseContext().getMetaData().getGroups().isEmpty()) {
+                                            // a root (all entries) GroupTreeNode
+                                            GroupTreeNode rootNode = getCurrentLibraryTab().getBibDatabaseContext()
+                                                                                           .getMetaData()
+                                                                                           .getGroups()
+                                                                                           .get()
+                                                                                           .copyNode();
+
+                                            // add rootNode to the Library if it does not have a root.
+                                            libraryTabCasted.getBibDatabaseContext()
+                                                                     .getMetaData()
+                                                                     .setGroups(rootNode);
+                                        }
+                                        Optional<GroupTreeNode> parentGroup = Optional.of(libraryTabCasted
+                                                .getBibDatabaseContext()
+                                                .getMetaData()
+                                                .getGroups().get());
+
+                                        for (String pathToSource : groupPathToSources) {
+                                            List<BibEntry> allEntries = getCurrentLibraryTab()
+                                                    .getBibDatabaseContext()
+                                                    .getEntries();
+                                            Optional<GroupTreeNode> groupTreeNodeToCopy = getCurrentLibraryTab()
+                                                    .getBibDatabaseContext()
+                                                    .getMetaData()
+                                                    .getGroups()
+                                                    .get()
+                                                    .getChildByPath(pathToSource);
+
+                                            // call the copy function
```

Comment:
I think you forgot this comment.

Reply:
I do not fully understand what do you mean by I forgot it, can you please clarify that?

Reply:
The comment doesn't add any value to the code. Anyone who knows Java should understand that the next line is calling the copy method :^). Generally you want to use comments to explain things that you can't express fluently in the code through the available syntax and good variable naming practice.

The reason behind avoiding comments when possible, is that comments tend to go out of date without any warning. For instance, if you change a method name, you will get compilations errors in all places where the method was used. You won't get that with comments, which can risk passing the wrong information to code readers.